maven-core/src/main/java/org/apache/maven/execution/MavenExecutionRequest.java

```diff
@@ -494,14 +504,51 @@ public interface MavenExecutionRequest {
 
     /**
      * @since 3.3.0
+     * @deprecated use {@link #setRootdir(Path)} instead
      */
+    @Deprecated
     void setMultiModuleProjectDirectory(File file);
 
     /**
      * @since 3.3.0
+     * @deprecated use {@link #getRootdir()} instead
      */
+    @Deprecated
     File getMultiModuleProjectDirectory();
 
+    /**
+     * Sets the root dir of the project.
+     *
+     * @since 4.0.0
+     */
+    MavenExecutionRequest setRootdir(Path rootdir);
+
+    /**
+     * Gets the root directory of the project, which is the parent directory containing the {@code .mvn} directory
+     * or a {@code pom.xml} file with the {@code root="true"} attribute.
+     * If there's no such file, an {@code IllegalStateException} will be thrown.
+     *
+     * @throws IllegalStateException if the root directory could not be found
+     * @since 4.0.0
+     */
+    Path getRootdir() throws IllegalStateException;
```

Comment:
That's a runtime exception and redundant in the signature.

Reply:
It's not mandatory, but definitely not redundant. That's documentation ;-)

Reply:
The documentation is already present in Javadoc. We don't declare REs anywhere, but Javadoc. It should be consistent throughout.

Reply:
Fixed